server/server.go

```diff
@@ -0,0 +1,100 @@
+package server
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+
+	"github.com/hashicorp/vault/api"
+)
+
+// New returns an unstarted HTTP server with health and proxy handlers.
+func New(logger *log.Logger, vaultClient *api.Client) *http.Server {
+	mux := http.ServeMux{}
+	mux.HandleFunc("/_health", healthHandler)
+	mux.HandleFunc("/", proxyHandler(logger, vaultClient))
+	srv := http.Server{
+		Addr:    "127.0.0.1:8200",
+		Handler: &mux,
+	}
+
+	return &srv
+}
+
+// The proxyHandler is based on the Send function from Vault Agent's proxy:
+// https://github.com/hashicorp/vault/blob/22b486b651b8956d32fb24e77cef4050df7094b6/command/agent/cache/api_proxy.go
+func proxyHandler(logger *log.Logger, vaultClient *api.Client) func(http.ResponseWriter, *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		switch r.Method {
+		case "GET":
+			cl, err := vaultClient.Clone()
+			if err != nil {
+				http.Error(w, err.Error(), 500)
+				return
+			}
+			cl.SetToken(vaultClient.Token())
+
+			body, err := ioutil.ReadAll(r.Body)
+			if err != nil {
+				http.Error(w, err.Error(), 500)
+				return
+			}
+
+			logger.Printf("Proxying %s %s\n", r.Method, r.URL.Path)
+			resp, err := proxyRequest(cl, r, body)
+			if err != nil {
+				http.Error(w, err.Error(), 502)
+				return
+			}
+
+			err = resp.Write(w)
+			if err != nil {
+				http.Error(w, "failed to write response back to requester", 500)
+				return
+			}
+			logger.Printf("Successfully proxied %s %s\n", r.Method, r.URL.Path)
+		default:
```

Comment:
Out of curiosity, why only GETs?

Reply:
We were taking a cautious approach to what we would support, but tbh it is a very artificial limitation, so I've removed it.